### DISCUSSION TITLE
Training on SLURM with multiple GPUs

### DISCUSSION THREAD

**Discussion by matval** [Lightning-AI/pytorch-lightning#17963] [DDP / multi-GPU / multi-node]
Hi,
I'm trying to train a model using Pytorch Lightining version 1.9.5 with DDPStrategy and use 2x V100.

If I run sbatch as:
```
#!/bin/bash

#SBATCH --nodes=1
#SBATCH --gres=gpu:v100:2
#SBATCH --ntasks-per-node=2
#SBATCH --mem-per-cpu=1200
#SBATCH --time=12:00:00

srun python3 train.py
```
The code runs fine, but it's using only two CPUs for data loading, which makes the training too slow. What is the correct way to enable the training on multiple GPUs with a different number of CPUs?
Thank you.

**Comment by changspencer** [comment]
Shouldn't you have the option to specify `--cpus-per-task` in your SLURM batch script? That's where I normally specify a higher number of CPUs for multiple workers.

**Comment by GuangShuaiWang** [comment]
Add` #SBATCH --ntasks-per-node=<gpu_num>` in slumr script, it work for me

**Reply by anthonyylee** [reply]
Doesn't the flag `--ntasks-per-node` specify the number of CPU per node? Or does it set the number of GPUs when one is running on a node with GPU?
